test/hotspot/jtreg/compiler/lib/ir_framework/test/IREncodingPrinter.java

```diff
@@ -176,6 +203,62 @@ private boolean hasAllRequiredFlags(String[] andRules, String ruleType) {
         return returnValue;
     }
 
+    private boolean hasRequiredFeaturesAnd(String[] andRules, String ruleType) {
```

Comment:
`ruleType` is always `applyIfTargetFeatureAnd` and can be replaced as such.

Suggestion for method name: `hasAllRequiredTargetFeatures()` to follow the existing naming scheme.